cxmon/src/mon_cmd.cpp

```diff
@@ -303,6 +307,288 @@ void binary_dump(void)
+		mon_error("Too many arguments");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it;
+	// Save break point
+	if((it = disabled_break_points.find(address)) == disabled_break_points.end())
+		active_break_points.insert(address);
+	else {
+		disabled_break_points.erase(it);
+		active_break_points.insert(address);
+	}
+}
+
+
+/*
+ * Remove Break Point
+ */
+void break_point_remove(void)
+{
+	uintptr index;
+
+	if (mon_token == T_END ||
+			!mon_expression(&index)) {
+		fprintf(monerr, "Expect index number of break point in hexadecimal.\n");
+		return;
+	}
+
+	if (mon_token != T_END) {
+		mon_error("Too many arguments");
+		return;
+	}
+
+	if (index > active_break_points.size()) {
+		mon_error("Illegal Index Number!");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it;
+
+	if (0 == index) {
+		active_break_points.clear();
+		printf("Removed all break points!\n");
+		return;
+	}
+
+	int pos = 1;
+	for (it = active_break_points.begin(); it != active_break_points.end(); it++)
+		if (pos++ == index)
+			break;
+	// Remove break point
+	printf("Removed break point %4x at address %08lx\n", index, *it);
+	active_break_points.erase(it);
+}
+
+
+/*
+ * Disable Break Point
+ */
+void break_point_disable(void)
+{
+	uintptr index;
+
+	if (mon_token == T_END ||
+			!mon_expression(&index)) {
+		fprintf(monerr, "Expect index number of break point in hexadecimal.\n");
+		return;
+	}
+
+	if (mon_token != T_END) {
+		mon_error("Too many arguments");
+		return;
+	}
+
+	if (index > active_break_points.size()) {
+		mon_error("Illegal Index Number!");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it;
+
+	if (0 == index) {
+		for (it = active_break_points.begin(); it != active_break_points.end(); it++)
+			disabled_break_points.insert(*it);
+		active_break_points.clear();
+		printf("Disabled all break points!\n");
+		return;
+	}
+
+	int pos = 1;
+	for (it = active_break_points.begin(); it != active_break_points.end(); it++)
+		if (pos++ == index)
+			break;
+	// Add to disable break points
+	printf("Disabled break point %4x at address %08lx\n", index, *it);
+	disabled_break_points.insert(*it);
+	// Remove break point
+	active_break_points.erase(it);
+}
+
+
+/*
+ * Enable Break Point
+ */
+void break_point_enable(void)
+{
+	uintptr index;
+
+	if (mon_token == T_END ||
+			!mon_expression(&index)) {
+		fprintf(monerr, "Expect index number of break point in hexadecimal.\n");
+		return;
+	}
+
+	if (mon_token != T_END) {
+		mon_error("Too many arguments");
+		return;
+	}
+
+	if (index > disabled_break_points.size()) {
+		mon_error("Illegal Index Number!");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it;
+
+	if (0 == index) {
+		for (it = disabled_break_points.begin(); it != disabled_break_points.end(); it++)
+			active_break_points.insert(*it);
+		disabled_break_points.clear();
+		printf("Enabled all break points!\n");
+		return;
+	}
+
+	int pos = 1;
+	for (it = disabled_break_points.begin(); it != disabled_break_points.end(); it++)
+		if (pos++ == index)
+			break;
```

Comment:
Can you replace this loop with std::advance() ?

Reply:
Good idea! Done!